### ISSUE TITLE
Corporation Product Progress >= 100 affecting multiplier?

### ISSUE THREAD

**Issue by Martin-Pitt** [danielyxie/bitburner#3737]
Although the description on progress specifies 0-100
https://github.com/danielyxie/bitburner/blob/acd796f0edfb6a1a8e642b5a05db6609dc4ca106/src/Corporation/Product.ts#L53

Due to there being no cap on progress going beyond 100 except for the check on industry
https://github.com/danielyxie/bitburner/blob/dev/src/Corporation/Industry.ts#L1022-L1024
https://github.com/danielyxie/bitburner/blob/acd796f0edfb6a1a8e642b5a05db6609dc4ca106/src/Corporation/Product.ts#L126

There is a possibility that progrMult can be >1.0
https://github.com/danielyxie/bitburner/blob/dev/src/Corporation/Product.ts#L134

Also is progrMult an artifact of old code? If a product is finished there is no potential for partial progress.

**Comment by phyzical** [comment]
@Martin-Pitt well spotted, i attempted to follow its origins but it goes back further than before the TS refactor (~2 years ago). im going to guess as you assume at one point you could create products before they were finished.

i think we are safe to remove progrMult entirely imo.

the only thing i cant think it is still doing is that its gradually increasing the stats of the product as its created.

The safest option would be to cap the progrMult at a max of 1 though

**Comment by Martin-Pitt** [comment]
Some food for thought: If you do however keep it, one potential usecase is having a feature for tracking the ratings as it's being made -- which would allow for the user to tune their employee ratios. That could make it less surprising or frustrating -- I don't know what the intended game design is, so up to you.

(For context I was actually in the process of writing a script that I could feed different ratios to to figure out what the ratings might be and that was when I spotted that weird issue with the progress multiplier)

**Comment by phyzical** [comment]
yeah that's the only usecase i can see too tbh, but are these stats visible via the ui while being made? or is it api only?

**Comment by Martin-Pitt** [comment]
When you hover over the rating on a product you can see each sub rating.

**Comment by phyzical** [comment]
true yeah, imo lets just ditch the progMult 👍
